### DISCUSSION TITLE
Which method is appropriate to open graph-image

### DISCUSSION THREAD

**Discussion by hyjoong** [vercel/next.js#52706] [App Router] [answered]
I looked at the official document to use `open graph-image` technology and saw two ways.

1. use  `@vercel/og package`
```javascript
import { ImageResponse } from '@vercel/og';
```
 
https://vercel.com/docs/concepts/functions/edge-functions/og-image-generation

2. use `{ImageResponse} from 'next/server'`

https://nextjs.org/docs/app/api-reference/file-conventions/metadata/opengraph-image

The `'import {ImageResponse} from 'next/server'` method seems to be suitable, but I wonder the difference between the two methods

I wonder if there is a better performance than the traditional method if I use the second method

The project I am working on is using the `app router` method in the `next.js 13.4` version

**Comment by icyJoseph** [comment, marked as answer]
I'd say, ImageResponse from next/server is a far more specialized and complete way to do open graph handling.

Using `vercel/og` is but one piece of the puzzle. To handle open graph, you needs meta tags as well.

Before `opengraph-image` was implemented, I made a Route handler that returns the image using `vercel/og`, and then had to go to my route's meta data and point to that Route, as well as provide other complementary data.

For example:

- https://github.com/icyJoseph/icyJoseph/blob/dev/src/app/og-image/%5Bslug%5D/route.tsx
- https://github.com/icyJoseph/icyJoseph/blob/dev/src/app/blog/%5Bslug%5D/page.tsx#L16

However, the `opengraph-image` convention, includes the necessary meta tags for that route as well.

- https://nextjs.org/docs/app/api-reference/file-conventions/metadata/opengraph-image

I'd say handling fonts is also a bit easier, and you can export image properties. With the `vercel/og` you can still pass properties to the returned image though, but not an `alt` property for instance.

So, I'd say, `@vercel/og` is a core building block, of a larger, production grade, solution such  as the `opengraph-image` file convention.

**Reply by hyjoong** [reply]
Thanks for good info. appreciate
